feature: Add multi architecture support

whatthefinemanual opened the pull request:
Changes enable using this action also for different architectures. 

Changes include: 

- Updated readme
- Actual code changes that allow to specify architecture ( default is still amd64 ) 
- Updated action.yaml to set additional input with it's description and default value

whatthefinemanual committed:
feature: Add multi architecture support

whatthefinemanual committed:
Merge pull request #1 from whatthefinemanual/feature/enable-multi-architecture-support

feature: Add multi architecture support

dev-hanz-ops reviewed (commented):
Thanks for this useful addition to the action. Please consider the comments below.

whatthefinemanual committed:
feature: Pick up arch from host

Updated readme and actual code to use process.arch and based on that downloads actual binary

whatthefinemanual committed:
fix : cleanup action.yaml

whatthefinemanual committed:
Merge pull request #2 from whatthefinemanual/feature/enable-multi-architecture-support

feature:enable multi architecture support

dev-hanz-ops commented on src/run.js:
```diff
@@ -25,10 +25,11 @@ async function getGhCli(version) {
-  const downloadUrl = `https://github.com/cli/cli/releases/download/v${version}/gh_${version}_linux_amd64.tar.gz`;
-  console.log(`downloading ${downloadUrl}`);
 
+  let architecture = process.arch
```
I'm afraid this will not work. `process.arch` returns x64 when we need amd64. you can give it another shot using below pseudocode

```
let architecture = "amd64"
if process.arch in ["arm", "arm64"] {
  architecture = process.arch
}
```

can you please test the changes using at least one amd64 and one arm64 machine?

whatthefinemanual committed:
feature: Add correct logic to pick up arm64 or amd64 architecture (#3)

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

whatthefinemanual committed:
fix: Update README.md that supports only amd64 and arm64 architectures (#4)

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

* feature: Add correct logic to pick up arm64 or amd64 architecture

* fix: Update README.md that supports only amd64 and arm64 architectures

dev-hanz-ops commented:
Sorry mate, it went lost after the holidays. Merging now
